src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -537,77 +693,28 @@ def write_segmentation(
+                ),
             )
-            ps_metadata = metadata["Ophys"]["ImageSegmentation"]["plane_segmentations"][0]
-            if ps_metadata["name"] not in image_segmentation.plane_segmentations:
-                ps_exist = False
-            else:
-                ps = image_segmentation.get_plane_segmentation(ps_metadata["name"])
-                ps_exist = True
-            roi_ids = segext_obj.get_roi_ids()
-            accepted_list = segext_obj.get_accepted_list()
-            accepted_list = [] if accepted_list is None else accepted_list
-            rejected_list = segext_obj.get_rejected_list()
-            rejected_list = [] if rejected_list is None else rejected_list
-            accepted_ids = [1 if k in accepted_list else 0 for k in roi_ids]
-            rejected_ids = [1 if k in rejected_list else 0 for k in roi_ids]
-            roi_locations = np.array(segext_obj.get_roi_locations()).T
-
-            def image_mask_iterator():
-                for id in segext_obj.get_roi_ids():
-                    img_msks = segext_obj.get_roi_image_masks(roi_ids=[id]).T.squeeze()
-                    yield img_msks
-
-            if not ps_exist:
-                from hdmf.common import VectorData
-
-                input_kwargs.update(
-                    **ps_metadata,
-                    columns=[
-                        VectorData(
-                            data=H5DataIO(
-                                DataChunkIterator(image_mask_iterator(), buffer_size=buffer_size),
-                                compression=True,
-                                compression_opts=9,
-                            ),
-                            name="image_mask",
-                            description="image masks",
-                        ),
-                        VectorData(
-                            data=roi_locations,
-                            name="RoiCentroid",
-                            description="x,y location of centroid of the roi in image_mask",
-                        ),
-                        VectorData(
-                            data=accepted_ids,
-                            name="Accepted",
-                            description="1 if ROi was accepted or 0 if rejected as a cell during segmentation operation",
-                        ),
-                        VectorData(
-                            data=rejected_ids,
-                            name="Rejected",
-                            description="1 if ROi was rejected or 0 if accepted as a cell during segmentation operation",
-                        ),
-                    ],
-                    id=roi_ids,
-                )
 
-                ps = image_segmentation.create_plane_segmentation(**input_kwargs)
+            plane_segmentation_metadata = metadata["Ophys"]["ImageSegmentation"]["plane_segmentations"]
+            plane_segmentation_name = plane_segmentation_metadata[plane_no_loop]["name"]
```

Comment:
I am not sure about the `plane_no_loop` as a `plane_segmentation_metadata` key. If you look at the top of the function, the logic of the previous intent seems to be that the metadata of each of the segmentation extractor (corresponding each to an imaging plane) would be each a full metadata dictionary (hence they have the metadata as a list in lines 479.

In other words, instead of coding the metadata of each plane as an element of `metadata["Ophys"]["ImagSegmentation"]["plane_segmentations"]`. They intended to have a list of metadata (the full one) and the metadata of each plane being `metadata[plane_no_loop]["Ophys"]["ImageSegmentation"][0]`. 

Please double check me on that. I think this is very confusing and we should discuss it in the next group meeting.

Reply:
yeah, I'm not sure about this part either. I'll double check it.